Anytype/Sources/PresentationLayer/TextEditor/EditorPage/Views/Settings/Icon/ObjectIconPickerViewModel.swift

```diff
@@ -24,22 +24,34 @@ final class ObjectIconPickerViewModel: ObservableObject, ObjectIconPickerViewMod
     // MARK: - Private variables
     
     private let document: BaseDocumentGeneralProtocol
+    private let actionHandler: ObjectIconActionHandlerProtocol
     private var subscription: AnyCancellable?
-    private let onIconAction: (ObjectIconPickerAction) -> Void
         
     // MARK: - Initializer
     
-    init(
-        document: BaseDocumentGeneralProtocol,
-        onIconAction: @escaping (ObjectIconPickerAction) -> Void
-    ) {
+    init(document: BaseDocumentGeneralProtocol, actionHandler: ObjectIconActionHandlerProtocol) {
         self.document = document
-        self.onIconAction = onIconAction
+        self.actionHandler = actionHandler
         subscription = document.syncPublisher.sink { [weak self] in
             self?.updateState()
         }
     }
     
+    
+    func setEmoji(_ emojiUnicode: String) {
+        actionHandler.handleIconAction(document: document, action: .setIcon(.emoji(emojiUnicode: emojiUnicode)))
```

Comment:
Now screen contains uploading logic. Not HeaderViewModel. Header just handle uploading state from ObjectHeaderUploadingServiceProtocol